### PR TITLE
Use the automatically generated yaml files for manifest

### PR DESCRIPTION
**Pull request by heypnus**
This patch is syncing the latest NCP yamls to operator,
and merging multiple yaml files to one for every platform.

### CODE REVIEW

File: deploy/kubernetes/role.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@ rules:
   resources: [deployments, endpoints, ingressclasses, nodes, pods/log, replicationcontrollers, secrets]
   verbs: [get, list, watch]
 - apiGroups: [vmware.com]
-  resources: [loadbalancers, loadbalancers/status, nsxlbmonitors, nsxlbmonitors/status]
+  resources: [loadbalancers, loadbalancers/status, nsxlbmonitors, nsxlbmonitors/status, virtualnetworkinterfaces, virtualnetworkinterfaces/status, virtualnetworks, virtualnetworks/status]
```
Comment (dantingl) [inline deploy/kubernetes/role.yaml:42]: They are for WCP CRDs, do you think we should add them?

File: manifest/openshift4/coreos/ncp-openshift4.yaml [hunk 2]
```diff
@@ -0,0 +1,1231 @@
+
```
Comment (dantingl) [inline manifest/openshift4/coreos/ncp-openshift4.yaml:1]: If we only support Openshift on CoreOS, no need to add sub-directory under manifest/openshift4?

File: deploy/kubernetes/role.yaml (hunk 1 above)
Comment (heypnus) [reply deploy/kubernetes/role.yaml:42]: In NCP yaml the logic is:
```
    142    {% if nsxapi == "policy" %}
    143      - virtualnetworks
    144      - virtualnetworks/status
    145      - virtualnetworkinterfaces
    146      - virtualnetworkinterfaces/status
    147    {% endif %}
```
Looks that it's not only for WCP.

File: manifest/openshift4/coreos/ncp-openshift4.yaml (hunk 2 above)
Comment (heypnus) [reply manifest/openshift4/coreos/ncp-openshift4.yaml:1]: Just thought maybe we will support more OS for openshift in the future...

**Review by dantingl — APPROVED** [review]

**Pull request merged**